cassiopeia/tools/parameter_estimators.py

```diff
@@ -0,0 +1,426 @@
+    probability as `stochastic_missing_probability` in `tree.parameters`.
+    If they are not provided on the tree, then they may be provided as
+    function arguments. If neither or both parameters are provided by either of
+    these methods, the function errors.
+
+    In estimating the heritable missing rate from the stochastic missing data
+    probability, we take the proportion of stochastic missing data in the
+    character matrix as equal to the stochastic probability. Then using the
+    total observed proportion of missing data as well as the estimated
+    proportion of stochastic missing data we can estimate the proportion
+    of heritable missing data using the expression above. Finally, we use the
+    heritable proportion as an estimate of the probability a lineage acquires
+    a missing data event by the end of the phylogeny, and using this
+    probability we can estimate the rate.
+
+    In the case where the rate is per-generation (probability a heritable
+    missing data event occurs on an edge), it is estimated using:
+
+        heritable missing proportion =
+            1 - (1 - heritable missing rate) ^ (average depth of tree)
+
+    In the case where the rate is continuous, it is estimated using:
+
+        heritable_missing_proportion =
+            ExponentialCDF(average time of tree, heritable missing rate)
+
+    Note that these naive estimates perform better when the tree is ultrametric
+    in depth or time. The average depth/lineage time of the tree is used as a
+    proxy for the depth/total time when the tree is not ultrametric.
+
+    In calculating the heritable proportion from the heritable missing rate,
+    we need to consider whether to assume an implicit root. This is specified
+    by `assume_root_implicit_branch`. In the case where the tree does not have
+    a single leading edge from the root representing the progenitor cell before
+    cell division begins, this additional edge is added to the total time in
+    calculating the estimate if `assume_root_implicit_branch` is True.
+
+    In estimating the stochastic missing probability from the heritable missing
+    rate, we calculate the expected proportion of heritable missing data using
+    the heritable rate in the same way, and then as above use the total
+    proportion of missing data to estimate the stochastic proportion, which we
+    assume is equal to the probability.
+
+    Args:
+        tree: The CassiopeiaTree specifying the tree and the character matrix
+        continuous: Whether to calculate a continuous missing rate,
+            accounting for branch lengths. Otherwise, calculates a
+            discrete missing rate based on the number of generations
+        assume_root_implicit_branch: Whether to assume that there is an
+            implicit branch leading from the root, if it doesn't exist
+        stochastic_missing_probability: The stochastic missing probability.
+            Will override the value on the tree
```

Comment:
Can you give people an idea of what a reasonable `stochastic_missing_probability` is? Say 10%?